### PR TITLE
test: use codecov/codecov-action

### PR DESCRIPTION
**Pull request by Layouwen**
https://github.com/ant-design/ant-design/issues/50387

### CODE REVIEW

**Comment by afc163** [comment]
See https://github.com/react-component/table/pull/1173

**Comment by Layouwen** [comment]
这块 npm i lock only 会报没有 lock, 去掉后 npm ci 会报 lock 不一致.
这个该怎么处理 @afc163

File: .github/workflows/react-component-ci.yml [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
       - name: cache package-lock.json
-        uses: actions/cache@v2
+        uses: actions/cache@v4
         with:
           path: package-temp-dir
           key: lock-${{ github.sha }}
 
       - name: create package-lock.json
-        run: npm i --package-lock-only
+        run: npm i
```
Comment (afc163) [inline .github/workflows/react-component-ci.yml:23]:
```suggestion
        run: npm i --package-lock-only
```

File: .github/workflows/react-component-ci.yml [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
       - name: cache package-lock.json
-        uses: actions/cache@v2
+        uses: actions/cache@v4
         with:
           path: package-temp-dir
           key: lock-${{ github.sha }}
 
       - name: create package-lock.json
-        run: npm i --package-lock-only
+        run: npm i --package-lock-only --ignore-scripts
```
Comment (afc163) [inline .github/workflows/react-component-ci.yml:23]:
```suggestion
        run: npm i --package-lock-only --ignore-scripts --legacy-peer-deps
```

File: .github/workflows/react-component-ci.yml [hunk 3]
```diff
@@ -35,78 +31,85 @@ jobs:
...
         id: node_modules_cache_id
-        uses: actions/cache@v2
+        uses: actions/cache@v4
         with:
           path: node_modules
           key: node_modules-${{ hashFiles('**/package-temp-dir/package-lock.json') }}
 
       - name: install
         if: steps.node_modules_cache_id.outputs.cache-hit != 'true'
         run: npm ci
```
Comment (afc163) [inline .github/workflows/react-component-ci.yml:41]:
```suggestion
        run: npm ci --legacy-peer-deps
```

**Pull request merged**